state/machine_test.go

```diff
@@ -748,6 +748,63 @@ func (s *MachineSuite) TestMachineInstanceIdBlank(c *gc.C) {
 	c.Assert(string(iid), gc.Equals, "")
 }
 
+func (s *MachineSuite) TestAllSpacesNone(c *gc.C) {
+	machine, err := s.State.AddMachine("quantal", state.JobHostUnits)
+	spaces, err := machine.AllSpaces()
+	c.Assert(err, jc.ErrorIsNil)
+	c.Check(spaces.SortedValues(), gc.DeepEquals, []string{})
```

Comment:
I could have done:
c.Check(spaces, gc.DeepEquals, set.NewStrings())

But
 a) consistency
 b) easier to read a list of strings
 c) usually better comparisons (easier to look at a mismatched list and figure out what is wrong)